runtime/ops/otel.rs

```diff
@@ -0,0 +1,582 @@
+        tokio::spawn(task);
+      }
+    });
+  });
+
+  spawn_task_tx
+}
+
+#[derive(Clone, Copy)]
+struct OtelSharedRuntime;
+
+impl opentelemetry_sdk::runtime::Runtime for OtelSharedRuntime {
+  type Interval = Pin<Box<dyn Stream<Item = ()> + Send + 'static>>;
+  type Delay = Pin<Box<tokio::time::Sleep>>;
+
+  fn interval(&self, period: Duration) -> Self::Interval {
+    stream::repeat(())
+      .then(move |_| tokio::time::sleep(period))
+      .boxed()
+  }
+
+  fn spawn(&self, future: BoxFuture<'static, ()>) {
+    (*OTEL_SHARED_RUNTIME_SPAWN_TASK_TX)
+      .unbounded_send(future)
+      .expect("failed to send task to shared OpenTelemetry runtime");
+  }
+
+  fn delay(&self, duration: Duration) -> Self::Delay {
+    Box::pin(tokio::time::sleep(duration))
+  }
+}
+
+impl opentelemetry_sdk::runtime::RuntimeChannel for OtelSharedRuntime {
+  type Receiver<T: Debug + Send> = BatchMessageChannelReceiver<T>;
+  type Sender<T: Debug + Send> = BatchMessageChannelSender<T>;
+
+  fn batch_message_channel<T: Debug + Send>(
+    &self,
+    capacity: usize,
+  ) -> (Self::Sender<T>, Self::Receiver<T>) {
+    let (batch_tx, batch_rx) = tokio::sync::mpsc::channel::<T>(capacity);
+    (batch_tx.into(), batch_rx.into())
+  }
+}
+
+#[derive(Debug)]
+pub struct BatchMessageChannelSender<T: Send> {
+  sender: tokio::sync::mpsc::Sender<T>,
+}
+
+impl<T: Send> From<tokio::sync::mpsc::Sender<T>>
+  for BatchMessageChannelSender<T>
+{
+  fn from(sender: tokio::sync::mpsc::Sender<T>) -> Self {
+    Self { sender }
+  }
+}
+
+impl<T: Send> opentelemetry_sdk::runtime::TrySend
+  for BatchMessageChannelSender<T>
+{
+  type Message = T;
+
+  fn try_send(
+    &self,
+    item: Self::Message,
+  ) -> Result<(), opentelemetry_sdk::runtime::TrySendError> {
+    self.sender.try_send(item).map_err(|err| match err {
+      tokio::sync::mpsc::error::TrySendError::Full(_) => {
+        opentelemetry_sdk::runtime::TrySendError::ChannelFull
+      }
+      tokio::sync::mpsc::error::TrySendError::Closed(_) => {
+        opentelemetry_sdk::runtime::TrySendError::ChannelClosed
+      }
+    })
+  }
+}
+
+pub struct BatchMessageChannelReceiver<T> {
+  receiver: tokio::sync::mpsc::Receiver<T>,
+}
+
+impl<T> From<tokio::sync::mpsc::Receiver<T>>
+  for BatchMessageChannelReceiver<T>
+{
+  fn from(receiver: tokio::sync::mpsc::Receiver<T>) -> Self {
+    Self { receiver }
+  }
+}
+
+impl<T> Stream for BatchMessageChannelReceiver<T> {
+  type Item = T;
+
+  fn poll_next(
+    mut self: Pin<&mut Self>,
+    cx: &mut Context<'_>,
+  ) -> Poll<Option<Self::Item>> {
+    self.receiver.poll_recv(cx)
+  }
+}
+
+fn otel_create_globals(
+  config: OtelConfig,
+  op_state: &mut OpState,
+) -> anyhow::Result<()> {
+  // Parse the `OTEL_EXPORTER_OTLP_PROTOCOL` variable. The opentelemetry_*
+  // crates don't do this automatically.
+  // TODO(piscisaureus): enable GRPC support.
+  let protocol = match env::var("OTEL_EXPORTER_OTLP_PROTOCOL").as_deref() {
```

Comment:
For a follow-up PR: we should have a way to specify OTLP protocol/endpoint/attributes via arguments rather than environment variables.

Reply:
Is there a particular reason for that? I think having ability to configure using env vars and maybe in `deno.json` would be enough?

Reply:
I think only env vars is perfectly fine for now. Maybe something for a follow up if there is demand

Reply:
This can be replaced with `ExportConfig::default().protocol`.

Reply:
> Is there a particular reason for that? I think having ability to configure using env vars and maybe in `deno.json` would be enough?

The purpose would be to allow platforms (e.g. Deploy) to configure OTEL without exposing the internal configuration to the app.

Reply:
> This can be replaced with `ExportConfig::default().protocol`.

It doesn't - opentelemetry-rust doesn't read the `OTEL_EXPORTER_OTLP_PROTOCOL` environment variable. It uses a fixed default based on enabled crate features.